### PR TITLE
Additional seq io

### PR DESCRIPTION
**Pull request by AGanesan12**
added additional methods to write Fasta and Fastq files

### CODE REVIEW

File: src/main/kotlin/biokotlin/seqIO/FastqIO.kt [hunk 1]
```diff
@@ -132,5 +135,29 @@ $filename  Error: ${e.message}""")
...
+    bufferedWriter(newFilename).use { outputFile ->
+        for (record in input) {
+            val seq = when (record) {
+                is NucSeqRecord -> record.sequence.toString()
+                else -> throw IllegalStateException("writeFastq trying to output something other than Nuc Seq.")
+            }
+            outputFile.write("${record.id}\n")
+            outputFile.write("${seq}\n")
+            outputFile.write("+\n")
+            outputFile.write(record.letterAnnotations["quality"]?.joinToString("")?:"")
```
Comment (lynnjo) [inline src/main/kotlin/biokotlin/seqIO/FastqIO.kt:159]: This looks like the code writes an empty string if "letterAnnotations" is empty, which would be illegal for a fastQ file format.  In reality, the SeqRecord class has a "require" that verifies the annotation size matches the sequence size.

Consider added a 1 line command that indicates SeqRecord object ensures quality annotation size is the same as sequence size.

**Review by zrm22 — APPROVED** [review]

**Review by lynnjo — APPROVED** [review]

File: src/main/kotlin/biokotlin/seqIO/FastaIO.kt [hunk 2]
```diff
@@ -177,3 +178,27 @@ fun writeFasta(input: SequenceIterator, filename: String) {
...
+
+    bufferedWriter(newFilename).use { outputFile ->
+        for (record in input) {
+            val seq = when (record) {
+                is NucSeqRecord -> record.sequence.toString()
+                is ProteinSeqRecord -> record.sequence.toString()
+                else -> throw IllegalStateException("writeFasta trying to output something other than Nuc or Protein Seq.")
+            }
+            outputFile.write(">${record.id}\n")
+            outputFile.write(seq)
```
Comment (tcasstevens) [inline src/main/kotlin/biokotlin/seqIO/FastaIO.kt:200]: Do we want to limit the length of the lines. Maybe like this?
seq
            .chunked(60)
            .forEach { chunk ->
                writer.write(chunk)
                writer.newLine()
            }

**Review by tcasstevens — APPROVED** [review]

File: src/main/kotlin/biokotlin/seqIO/FastaIO.kt (hunk 2 above)
Comment (AGanesan12) [reply src/main/kotlin/biokotlin/seqIO/FastaIO.kt:200]: Thank you for your suggestion, but could you clarify why limiting the length of the lines is necessary?

File: src/main/kotlin/biokotlin/seqIO/FastqIO.kt (hunk 1 above)
Comment (AGanesan12) [reply src/main/kotlin/biokotlin/seqIO/FastqIO.kt:160]: Done. Thank you!

File: src/main/kotlin/biokotlin/seqIO/FastaIO.kt (hunk 2 above)
Comment (tcasstevens) [reply src/main/kotlin/biokotlin/seqIO/FastaIO.kt:200]: My suggested code breaks up the sequence into 60 character chunks instead of having one long line

Comment (AGanesan12) [reply src/main/kotlin/biokotlin/seqIO/FastaIO.kt:200]: That makes sense, I updated the program so that it breaks up the sequence into 60 character chunks

**Pull request merged**